lib/Subscriber/Phpcr/RemoveSubscriber.php

```diff
@@ -55,12 +56,20 @@ public static function getSubscribedEvents()
      * remove any references to the node.
      *
      * @param RemoveEvent $event
+     *
+     * @throws DocumentReferencedException
      */
     public function handleRemove(RemoveEvent $event)
     {
         $document = $event->getDocument();
         $node = $this->documentRegistry->getNodeForDocument($document);
 
+        $references = iterator_to_array($node->getReferences());
```

Comment:
I tried to do that, but it was not possible due to a few issues in jackalope, and some architectural decisions you made in the document manager (e.g. the abstinence of a unit of work). The exception is not thrown before the `flush` event, and I found no way to get the affected nodes at this spot.